Source/Core/VideoCommon/AVIDump.cpp

```diff
@@ -392,7 +392,10 @@ void AVIDump::CheckResolution(int width, int height)
   // (possibly width as well, but no examples known) to have a value of zero. This can occur as the
   // VI is able to be set to a zero value for height/width to disable output. If this is the case,
   // simply keep the last known resolution of the video for the added frame.
-  if ((width != s_width || height != s_height) && (width > 0 && height > 0))
+  bool width_changed = g_Config.iDumpWidth ? (s_width != g_Config.iDumpWidth) : (width != s_width);
```

Comment:
My understanding is that this code is activated only while a game is running (I've only run into it when games switch resolutions at the very beginning, for some reason).

I don't know if you can change the general config while a game is running, but even if you can I don't see the use-case for changing `iDumpWidth`